endpointslice/endpointslice_controller.go

```diff
@@ -363,6 +369,9 @@ func (c *Controller) syncService(logger klog.Logger, key string) error {
 		c.triggerTimeTracker.DeleteService(namespace, name)
 		c.reconciler.DeleteService(namespace, name)
 		c.endpointSliceTracker.DeleteService(namespace, name)
+		if c.noServiceHook != nil {
```

Comment:
The comment says this is run when the svc is not found, but in that case this will return above if `apierrors.IsNotFound(err)`is true.

Reply:
Hmm the return above is an early return to directly return theerror when `apierrors.IsNotFound(err)` is `false` actually so I believe it should work